src/activity_handlers.cpp

```diff
@@ -2688,6 +2689,13 @@ void activity_handlers::mend_item_finish( player_activity *act, Character *you )
     for( const auto &[var_name, var_value] : fix.set_variables ) {
         target.set_var( var_name, var_value );
     }
+    for( const auto &[var_name, var_value] : fix.adjust_variables_multiply ) {
+        const std::string var_value_multiplier = var_value;
+        const std::string var_oldvalue = target.get_var( var_name );
+        const int var_oldvalue_int = std::stoi( var_oldvalue );
+        const double var_value_multiplier_double = std::stod( var_value_multiplier );
+        target.set_var( var_name, round( var_oldvalue_int * var_value_multiplier_double ) );
```

Comment:
If you're multiplying a value, presumably it's already a double?
```suggestion
        const double var_value_multiplier = var_value;
        const double var_oldvalue = target.get_var( var_name, 0.0 );
        target.set_var( var_name, std::round( var_oldvalue * var_value_multiplier ) );
```

Reply:
I think this change breaks the game. I have to look at this again.

Reply:
Please explain. How did it do so?

Reply:
It was giving me an error on game load and then kicking me back to the main screen. Want me to put those two changes back in, recomp, and get a copy of the error?

Reply:
I'd appreciate that! I think it's a preferable solution.

Reply:
Looking at your commits, my guess is the game load error is because the uses of this in JSON weren't adjusted to no longer be strings.

Reply:
```
 DEBUG    : Error: Json error: data/json/faults/fixes_gun.json:39:43: <color_white><color_cyan>Expected number</color>

    "skills": { "mechanics": 1 },
    "set_variables": { "rust_timer": "0" },
<color_light_red>    "adjust_variables_multiply": { "dirt": ".8" },</color>
                                         <color_cyan>▲▲▲</color>
    "requirements": [ [ "gun_cleaning_quick", 1 ] ],
    "time_save_profs": { "prof_gun_cleaning": 0.5 },
</color>


 FUNCTION : bool main_menu::load_game(const string&, const save_t&)
 FILE     : src/main_menu.cpp
 LINE     : 1103
 VERSION  : baa235478f-dirty

```

Reply:
Yep, exactly that. `"dirt": 0.8` (value is a number) is expected, not `"dirt": ".8"` (value is a string)